src/android/src/main/java/com/airbnb/android/react/lottie/LottieAnimationViewPropertyManager.java

```diff
@@ -18,131 +26,151 @@
-    this.loop = loop;
-  }
-
-  public void setUseHardwareAcceleration(boolean useHardwareAcceleration) {
-    this.useHardwareAcceleration = useHardwareAcceleration;
-  }
-
-  public void setScaleType(ImageView.ScaleType scaleType) {
-    this.scaleType = scaleType;
-  }
-
-  public void setImageAssetsFolder(String imageAssetsFolder) {
-    this.imageAssetsFolder = imageAssetsFolder;
-  }
-
-  public void setEnableMergePaths(boolean enableMergePaths) {
-    this.enableMergePaths = enableMergePaths;
-  }
-
-  /**
-   * Updates the view with changed fields.
-   * Majority of the properties here are independent so they are has to be reset to null
-   * as soon as view is updated with the value.
-   *
-   * The only exception from this rule is the group of the properties for the animation.
-   * For now this is animationName and cacheStrategy. These two properties are should be set
-   * simultaneously if the dirty flag is set.
-   */
-  public void commitChanges() {
-    LottieAnimationView view = viewWeakReference.get();
-    if (view == null) {
-      return;
+    public void setCacheStrategy(LottieAnimationView.CacheStrategy strategy) {
+        this.cacheStrategy = strategy;
+        this.animationNameDirty = true;
     }
 
-    if (animationJson != null) {
-      view.setAnimation(new JsonReader(new StringReader(animationJson)));
-      animationJson = null;
+    public void setProgress(Float progress) {
+        this.progress = progress;
     }
 
-    if (animationNameDirty) {
-      view.setAnimation(animationName, cacheStrategy);
-      animationNameDirty = false;
+    public void setSpeed(float speed) {
+        this.speed = speed;
     }
 
-    if (progress != null) {
-      view.setProgress(progress);
-      progress = null;
+    public void setLoop(boolean loop) {
+        this.loop = loop;
     }
 
-    if (loop != null) {
-      view.setRepeatCount(loop ? LottieDrawable.INFINITE : 0);
-      loop = null;
+    public void setUseHardwareAcceleration(boolean useHardwareAcceleration) {
+        this.useHardwareAcceleration = useHardwareAcceleration;
     }
 
-    if (speed != null) {
-      view.setSpeed(speed);
-      speed = null;
+    public void setScaleType(ImageView.ScaleType scaleType) {
+        this.scaleType = scaleType;
     }
 
-    if (useHardwareAcceleration != null) {
-      view.useHardwareAcceleration(useHardwareAcceleration);
-      useHardwareAcceleration = null;
+    public void setImageAssetsFolder(String imageAssetsFolder) {
+        this.imageAssetsFolder = imageAssetsFolder;
     }
 
-    if (scaleType != null) {
-      view.setScaleType(scaleType);
-      scaleType = null;
+    public void setEnableMergePaths(boolean enableMergePaths) {
+        this.enableMergePaths = enableMergePaths;
     }
 
-    if (imageAssetsFolder != null) {
-      view.setImageAssetsFolder(imageAssetsFolder);
-      imageAssetsFolder = null;
+    public void setColorFilterLayers(ArrayList<Map<String,String>> colorLayers) {
```

Comment:
Have accept a. `Map<String, Integer>` instead

Reply:
The color is coming in as a string